File: core/NumberOfLinesToWriteString/NumberOfLinesToWriteString.py
from typing import List

def numberOfLines(widths: List[int], s: str) -> List[int]:
    row, remain, line = 0, 0, 0
    for ch in s:
        if widths[ord(ch) - 97] + line > 100:
            line = widths[ord(ch) - 97]
            row += 1
        else:
            line += widths[ord(ch) - 97]

    if line > 0:
        row += 1

    return [row, line]

File: core/NumberOfLinesToWriteString/NumberOfLinesToWriteString.java
package com.dbc;

public class NumberOfLinesToWriteString {
    public int[] numberOfLines(int[] widths, String s) {
        int row = 0, line = 0;
        for (int i = 0; i < s.length(); i++) {
            if (widths[s.charAt(i) - 'a'] + line > 100) {
                line = widths[s.charAt(i) - 'a'];
                row++;
            } else {
                line += widths[s.charAt(i) - 'a'];
            }
        }

        if (line != 0) row++;
        return new int[]{row, line};
    }
}
